nixos/modules/services/x11/urserver.nix

```diff
@@ -0,0 +1,72 @@
+# urserver service
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.urserver;
+in {
+
+  options.services.urserver = {
+
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enable Unified Remote server.
+      '';
+    };
+
+    usedTCPPorts = mkOption {
+      type = types.listOf types.int;
+      default = [ 9510 9512 ];
+      description = ''
+        TCP ports that are used by urserver.
+        By default port 9510 is used for the web server.
+        By default port 9512 is used for WiFi connections.
+      '';
+    };
+
+    usedUDPPorts = mkOption {
+      type = types.listOf types.int;
+      default = [ 9511 9512 ];
+      description = ''
+        UDP ports that are used by urserver.
+        By default port 9511 is used for automatic server discovery.
+        By default port 9512 is used for fast connections.
+      '';
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [ pkgs.urserver ];
+
+    systemd.services.urserver = {
```

Comment:
Is running this services as `root` a hard requirement?

Reply:
I tried running it as a user service early on, but then the server doesn't have the permissions to interact with uinput, e.g. to control the mouse using the basic remote.

Reply:
Maybe `udev` rules, or some `systemd` configuration can make this possible.

Another option, if this application doesn't make sense to run as a desktop user, might be `DynamicUser`.

Without knowing details of the software it is hard to know what the best solution is, so take these suggestions as just that: suggestions.

Reply:
I went back to basics and looked at the logs again and I found that actually the server tries to use X11 methods first, and only wants uinput as a fallback. I've modified to package part to take the required buildInputs and runtime LD_LIBRARY_PATH so that it finds everything it needs without needing to run with elevated permissions.